x-pack/plugins/siem/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -189,17 +190,25 @@ export const searchAfterAndBulkCreate = async ({
         throttle,
       });
       logger.debug('finished next bulk index');
+      logger.debug(`created ${createdCount} signals`);
       toReturn.createdSignalsCount += createdCount;
       if (bulkDuration) {
         toReturn.bulkCreateTimes.push(bulkDuration);
       }
+
+      if (filteredEvents?.hits?.hits[0]?.sort == null) {
```

Comment:
I don't think you need all the `?`, optional chaining do you? Something isn't right between the types and the data if you do? I took them away and this worked:

```ts
if (filteredEvents.hits.hits[0].sort == null) {
```

Also, below you're using it without the `?`:

```ts
sortId = filteredEvents.hits.hits[0].sort[0];
```

Reply:
You are correct I don't need the optional chaining. That was leftover from before the null checking which was cleaned up with your ternary, so thanks for that 😄